pylint/checkers/modified_iterating_checker.py

```diff
@@ -81,7 +81,7 @@ def _modified_iterating_check(
                 msg_id = "modified-iterating-dict"
             elif isinstance(inferred, nodes.Set):
                 msg_id = "modified-iterating-set"
-        elif not isinstance(iter_obj, nodes.Name):
+        elif not isinstance(iter_obj, (nodes.Name, nodes.Attribute)):
```

Comment:
should be worth to think more about the `Attribute` case, should it raise a message in all cases?
maybe there is a false positive here?